crates/stackable-operator/src/commons/listener.rs

```diff
@@ -63,12 +63,22 @@ pub struct ListenerClassSpec {
     /// break Local mode (IONOS so far).
     #[serde(default = "ListenerClassSpec::default_service_external_traffic_policy")]
     pub service_external_traffic_policy: KubernetesTrafficPolicy,
+
+    /// Whether addresses should prefer using the IP address (`IP`) or the hostname (`Hostname`).
+    ///
+    /// The other type will be used if the preferred type is not available.
+    #[serde(default = "ListenerClassSpec::default_preferred_address_type")]
+    pub preferred_address_type: AddressType,
 }
 
 impl ListenerClassSpec {
     const fn default_service_external_traffic_policy() -> KubernetesTrafficPolicy {
         KubernetesTrafficPolicy::Local
     }
+
+    const fn default_preferred_address_type() -> AddressType {
+        AddressType::Hostname
+    }
```

Comment:
This is a change for NodePort and ClusterIP services, previously NodePort and ClusterIP services would use the node/service IP (respectively), while ListenerClasses would use the hostname.

Defaulting to the node hostname is arguably better for consistency, but requires nodes to have valid DNS hostnames registered.